test/SIL/ownership-verifier/borrow_extract_validate.sil

```diff
@@ -0,0 +1,223 @@
+// CHECK: Non Consuming User:   %7 = apply %6(%4) : $@convention(thin) (@guaranteed Klass) -> ()
+// CHECK: Block: bb0
+// CHECK-LABEL: Error#: 0. End Error in Function: 'borrowed_extract_dominate_incorrect'
+// CHECK-NOT: Error#: {{[0-9][0-9]*}}. End Error in Function: 'borrowed_extract_dominate_incorrect'
+sil [ossa] @borrowed_extract_dominate_incorrect : $@convention(thin) (@guaranteed WrapperStruct, @guaranteed WrapperStruct) -> () {
+bb0(%0 : @guaranteed $WrapperStruct, %1 : @guaranteed $WrapperStruct):
+  %copy0 = copy_value %0 : $WrapperStruct
+  %borrow0 = begin_borrow %copy0 : $WrapperStruct
+  %elem0 = struct_extract %borrow0 : $WrapperStruct, #WrapperStruct.cls
+  end_borrow %borrow0 : $WrapperStruct
+  %func = function_ref @blackhole_spl1 : $@convention(thin) (@guaranteed Klass) -> ()
+  %call1 = apply %func(%elem0) : $@convention(thin) (@guaranteed Klass) -> ()
+  destroy_value %copy0 : $WrapperStruct
+  %ret = tuple ()
+  return %ret : $()
+}
+
+// CHECK-NOT: Function: 'borrowed_extract_dominate_discrepency_simple_correct'
+sil [ossa] @borrowed_extract_dominate_discrepency_simple_correct : $@convention(thin) (@guaranteed WrapperStruct) -> () {
+bb0(%0 : @guaranteed $WrapperStruct):
+  %borrow0 = begin_borrow %0 : $WrapperStruct
+  %elem = struct_extract %borrow0 : $WrapperStruct, #WrapperStruct.cls
+  br bb1(%elem : $Klass)
+bb1(%newelem : @guaranteed $Klass):
+  end_borrow %borrow0 : $WrapperStruct
+  %res = tuple ()
+  return %res : $()
+}
+
+// CHECK-NOT: Function: 'simple_extract_as_phi_arg'
+sil [ossa] @simple_extract_as_phi_arg : $@convention(thin) (@guaranteed WrapperStruct) -> () {
+bb0(%0 : @guaranteed $WrapperStruct):
+  %elem = struct_extract %0 : $WrapperStruct, #WrapperStruct.cls
+  br bb1(%elem : $Klass)
+bb1(%newelem : @guaranteed $Klass):
+  %res = tuple ()
+  return %res : $()
+}
+
+// CHECK-LABEL: Error#: 0. Begin Error in Function: 'borrowed_extract_dominate_discrepency_incorrect1'
+// CHECK: Found outside of lifetime use?!
+// CHECK: Value: %7 = argument of bb1 : $WrapperStruct
+// CHECK: Consuming User:   end_borrow %7 : $WrapperStruct
+// CHECK: Non Consuming User:   %10 = apply %9(%6) : $@convention(thin) (@guaranteed Klass) -> ()
+// CHECK: Block: bb1
+// CHECK-LABEL: Error#: 0. End Error in Function: 'borrowed_extract_dominate_discrepency_incorrect1'
+// CHECK-NOT: Error#: {{[0-9][0-9]*}}. End Error in Function: 'borrowed_extract_dominate_discrepency_incorrect1'
+sil [ossa] @borrowed_extract_dominate_discrepency_incorrect1 : $@convention(thin) (@guaranteed WrapperStruct, @guaranteed WrapperStruct) -> () {
+bb0(%0 : @guaranteed $WrapperStruct, %1 : @guaranteed $WrapperStruct):
+  %copy0 = copy_value %0 : $WrapperStruct
+  %borrow0 = begin_borrow %copy0 : $WrapperStruct
+  %elem0 = struct_extract %borrow0 : $WrapperStruct, #WrapperStruct.cls
+  br bb1(%elem0 : $Klass, %borrow0 : $WrapperStruct)
+bb1(%newelem : @guaranteed $Klass, %newborrow : @guaranteed $WrapperStruct):
```

Comment:
We generally put a space in between bb1 and the block before it. Just makes it easier to read.